### DISCUSSION TITLE
Allow `Step` content to be placed in between stepper UI in `vertical` mode

### DISCUSSION THREAD

**Discussion by vinayakakv** [mantinedev/mantine#1060] [Feature requests]
Currently, in the `Stepper`, the content of the steps is rendered after the stepper UI.

While it is good with `orientation="horizontal"`, it does not provide a nice experience in smaller screen sizes.

With `orientation="vertical"`, if we support the content placed _in between_ the steps, it would provide a seamless experience, in which user does not have to navigate top and bottom to understand in which step they are currently in.

I copied the source file [`src/components/Stepper/Stepper.tsx`](https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/components/Stepper/Stepper.tsx) and dirty-fixed it to achieve the result in the screenshot.

<img width="642" alt="image" src="https://user-images.githubusercontent.com/24838519/159149165-58cf576b-7e81-4b67-9643-3b685853bbde.png">

If such support would be included in the package, for example, by means of a flag, that would be great!

The MUI stepper seems to have this functionality already - https://mui.com/components/steppers/#vertical-stepper

**Comment by rtivital** [comment]
I'm fine with adding this feature, I will not work on it myself though as I'm busy with other things

**Reply by vinayakakv** [reply]
Thanks for the reply @rtivital. How do you think is the appropriate way to implement it?
- Maybe adding this behaviour when `orientation="vertical"`? This would break existing uses
- Maybe adding a new prop `contentPlacement: "inside" | "outside"`? I think this is not applicable when `orientation="horizontal"`.

Currently, since I'm doing it in a _hacky_ way to get it working for my usecase, I did not add a new prop or anything.

**Reply by rtivital** [reply]
- It should look the same as Timeline component – https://mantine.dev/core/timeline/
- `contentPlacement` prop is fine
- It should be animated with Collapse component

**Reply by vinayakakv** [reply]
Got that, thanks 👍🏽 . Will try to set a development version of the library and send a PR, incorporating these features.

**Reply by hkkvo** [reply]
Would like to know any progress on it.

**Reply by rtivital** [reply]
No progress currently

**Comment by hkkvo** [comment]
We are in the middle of moving of our MUI components to Mantine because of the great features and we stumbled across the same issue of stepper mobile view. Is there any quick fix we can follow to mitigate this problem for now. it would be very helpful, we are going production with Mantine in two weeks.

**Reply by vinayakakv** [reply]
Hey @hkkvo, I had this issue and made a quick hack by editing the source code (and by placing it in my `src/` directory). I created a [gist](https://gist.github.com/vinayakakv/1e2a333eb1045088ff0f274fddf192f4) of it. Hope you find that helpful.

Sorry that I could not contribute back this to Mantine project. I would be grateful if anyone picks this up.

The implementation looked like this:
<img width="210" alt="image" src="https://user-images.githubusercontent.com/24838519/184796050-a33e6201-ff9e-462d-99a6-c9a559abec56.png">

**Reply by hkkvo** [reply]
Thank you @vinayakakv sharing the Git link

**Reply by hubertBS42** [reply]
Any chance you can help me with implementing this with @mantine/core 7.1.2?

**Comment by EmilioEsposito** [comment]
any update on this? It hinders Stepper usability on mobile. I'm trying to Migrate from MUI and used this feature a lot.

**Comment by hubertBS42** [comment]
Has there been any update on this? I tried implementing the solution @vinayakakv presented but quickly realized I don't know what I am doing lol.
